lib/puppet/metatype/manager.rb

```diff
@@ -143,29 +143,29 @@ def rmtype(name)
   #
   def type(name)
     # Avoid loading if name obviously is not a type name
-    if name.to_s.include?(':')
-      return nil
-    end
+    return nil if name.to_s.include? ':'
 
     @types ||= {}
 
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
i'm not sure i understood the comment correctly but ruby won't return if `@types[name]` is not truthy (or doesnt exist). setting `@types[name]` to nil wouldn't return without switching the condition to `#include?`. i'm sorry if i completely missed your point.